workers/unity/Packages/com.improbable.gdk.core/Utility/TaskUtility.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Improbable.Gdk.Core
+{
+    public static class TaskUtility
+    {
+        public static async Task<T> WithCancellation<T>(this Task<T> task, CancellationToken cancellationToken)
+        {
+            var tcs = new TaskCompletionSource<bool>();
+            using (cancellationToken.Register(s => ((TaskCompletionSource<bool>) s).TrySetResult(true), tcs))
+            {
+                if (task != await Task.WhenAny(task, tcs.Task))
+                {
+                    throw new OperationCanceledException(cancellationToken);
```

Comment:
We don't seem to catch this anywhere?

Reply:
it's getting caught in the `WorkerConnector`. Not specifically this exception, but it will just be propagated up to fail the whole worker creation